### PR TITLE
implement function to get single random Recipe from Database.

### PR DESCRIPTION
**Pull request by HendrikNeuschulz**

### CODE REVIEW

File: backend/src/test/java/com/github/hendrikneuschulz/backend/controller/RecipeControllerTest.java [hunk 1]
```diff
@@ -50,4 +41,24 @@ void testGetRecipes() {
...
+        when(recipeService.getRecipeList()).thenReturn(recipeList);
+
+
+        RecipeController recipeController = new RecipeController(recipeService);
+        Recipe randomRecipe = recipeController.getRandomRecipe();
+        assertNotNull(randomRecipe);
+
+
+        assertTrue(recipeList.contains(randomRecipe));
+    }
```
Comment (Kreyno93) [inline backend/src/test/java/com/github/hendrikneuschulz/backend/controller/RecipeControllerTest.java]: is this an intergration test (cause its in the Controller) or should it be a service test? Either way a new instance of controller isnt needed in both. propably wanted to instantiate the service with a mocked repo? for a unit test?

File: backend/src/main/java/com/github/hendrikneuschulz/backend/controller/RecipeController.java [hunk 2]
```diff
@@ -5,20 +5,30 @@
...
+    @GetMapping
     public List<Recipe> getRecipes() {
         return recipeService.getRecipeList();
     }
 
+    @GetMapping("/random")
+    public Recipe getRandomRecipe() {
+        List<Recipe> recipeList = recipeService.getRecipeList();
+        int randomIndex = new SecureRandom().nextInt(recipeList.size());
+        return recipeList.get(randomIndex);
```
Comment (valfink) [inline backend/src/main/java/com/github/hendrikneuschulz/backend/controller/RecipeController.java]: You could separate this logic into the service. Then you wouldn't have any business logic in the controller :)

File: backend/src/test/java/com/github/hendrikneuschulz/backend/controller/RecipeControllerTest.java [hunk 3]
```diff
@@ -50,4 +41,24 @@ void testGetRecipes() {
...
+        );
+        when(recipeService.getRecipeList()).thenReturn(recipeList);
+
+
+        RecipeController recipeController = new RecipeController(recipeService);
+        Recipe randomRecipe = recipeController.getRandomRecipe();
+        assertNotNull(randomRecipe);
+
+
+        assertTrue(recipeList.contains(randomRecipe));
```
Comment (valfink) [inline backend/src/test/java/com/github/hendrikneuschulz/backend/controller/RecipeControllerTest.java]: Nice test for a random function!

File: frontend/src/compontens/RandomRecipe.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import React, {useEffect, useState} from 'react';
+import axios from 'axios';
+import {Recipe} from '../model/Recipe';
+import RecipeItem from './RecipeItem';
+
+
+export default function RandomRecipe() {
+    const [recipe, setRecipe] = useState<Recipe | null>(null);
```
Comment (valfink) [inline frontend/src/compontens/RandomRecipe.tsx]: I once got the comment that `null` is not a good value in the TypeScript world and that it is better to user `undefined` and the internet seemed to confirm it, so I pass this on :)

File: frontend/src/logowtf.png
Comment (valfink) [inline frontend/src/logowtf.png:1]: Cool logo!

**Review by valfink — APPROVED** [review]
Nice work! 💪

**Pull request merged**